[Profiler] Remove `addMemoryUsageActivity` from kineto_shim

robieta opened the pull request:
Summary: `addMemoryUsageActivity` is effectively the same as `addCPUActivity` except it also has `addMetadata` calls. This change generalizes `addCPUActivity` (We have to define an enum to map to `libkineto::ActivityType`) to accept arbitrary metadata strings.

Test Plan: Unit tests

Differential Revision: D36070203

aaronenyeshi reviewed (approved):
Reviewed internally, LGTM.

robieta commented:
@pytorchbot rebase this please

Commit:
[Profiler] Remove `addMemoryUsageActivity` from kineto_shim (#76678)

Summary:
`addMemoryUsageActivity` is effectively the same as `addCPUActivity` except it also has `addMetadata` calls. This change generalizes `addCPUActivity` (We have to define an enum to map to `libkineto::ActivityType`) to accept arbitrary metadata strings.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/76678

Test Plan: Unit tests

Reviewed By: aaronenyeshi

Differential Revision: D36070203

Pulled By: robieta

fbshipit-source-id: 5d6ff65d16588ca1cc3d893dcfb960ea6249af58

robieta commented:
@pytorchbot merge this please
